File: IHaveATheory/app/src/main/java/com/avoid/ihaveatheory/model/SaveFile.java
package com.avoid.ihaveatheory.model;

public class SaveFile {
    //----------------------------------------------------------------------------------------------

    private String name = "default";
    private Scenario scenario = null;
    private Difficulty difficulty = null;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Scenario getScenario() {
        return scenario;
    }

    public void setScenario(Scenario scenario) {
        this.scenario = scenario;
    }

    public Difficulty getDifficulty() {
        return difficulty;
    }

    public void setDifficulty(Difficulty difficulty) {
        this.difficulty = difficulty;
    }

    //----------------------------------------------------------------------------------------------

    private Backpack backpack = null;

    public Backpack getBackpack() {
        return backpack;
    }

    public void setBackpack(Backpack backpack) {
        this.backpack = backpack;
    }

    //----------------------------------------------------------------------------------------------

    private int health = 0;
    private int stamina = 0;
    private int heat = 0;
    private int hunger = 0;
    private int thirst = 0;

    public int getHealth() {
        return health;
    }

    public void setHealth(int health) {
        if (health > difficulty.getMAX_HEALTH()) {
            this.health = difficulty.getMAX_HEALTH();
        } else if (health < 0) {
            this.health = 0;
        } else {
            this.health = health;
        }
    }

    public int getStamina() {
        return stamina;
    }

    public void setStamina(int stamina) {
        if (stamina > difficulty.getMAX_STAMINA()) {
            this.stamina = difficulty.getMAX_STAMINA();
        } else if (stamina < 0) {
            this.stamina = 0;
        } else {
            this.stamina = stamina;
        }
    }

    public int getHeat() {
        return heat;
    }

    public void setHeat(int heat) {
        if (heat > difficulty.getMAX_HEAT()) {
            this.heat = difficulty.getMAX_HEAT();
        } else if (heat < 0) {
            this.heat = 0;
        } else {
            this.heat = heat;
        }
    }

    public int getHunger() {
        return hunger;
    }

    public void setHunger(int hunger) {
        if (hunger > difficulty.getMAX_HUNGER()) {
            this.hunger = difficulty.getMAX_HUNGER();
        } else if (hunger < 0) {
            this.hunger = 0;
        } else {
            this.hunger = hunger;
        }
    }

    public int getThirst() {
        return thirst;
    }

    public void setThirst(int thirst) {
        if (thirst > difficulty.getMAX_THIRST()) {
            this.thirst = difficulty.getMAX_THIRST();
        } else if (thirst < 0) {
            this.thirst = 0;
        } else {
            this.thirst = thirst;
        }
    }

    //----------------------------------------------------------------------------------------------
}

File: app/src/main/java/com/avoid/ihaveatheory/model/SaveFile.kt
package com.avoid.ihaveatheory.model

import java.io.Serializable

class SaveFile : Serializable {

    var name = "default"
    var scenario: Scenario? = null
    var difficulty: Difficulty? = null

    var backpack: Backpack? = null

    var health = 0
        set(health) = if (health > difficulty!!.maX_HEALTH) {
            field = difficulty!!.maX_HEALTH
        } else if (health < 0) {
            field = 0
        } else {
            field = health
        }
    var stamina = 0
        set(stamina) = if (stamina > difficulty!!.maX_STAMINA) {
            field = difficulty!!.maX_STAMINA
        } else if (stamina < 0) {
            field = 0
        } else {
            field = stamina
        }
    var heat = 0
        set(heat) = if (heat > difficulty!!.maX_HEAT) {
            field = difficulty!!.maX_HEAT
        } else if (heat < 0) {
            field = 0
        } else {
            field = heat
        }
    var hunger = 0
        set(hunger) = if (hunger > difficulty!!.maX_HUNGER) {
            field = difficulty!!.maX_HUNGER
        } else if (hunger < 0) {
            field = 0
        } else {
            field = hunger
        }
    var thirst = 0
        set(thirst) = if (thirst > difficulty!!.maX_THIRST) {
            field = difficulty!!.maX_THIRST
        } else if (thirst < 0) {
            field = 0
        } else {
            field = thirst
        }

}
